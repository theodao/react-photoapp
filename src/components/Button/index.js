import React from 'react';
import { Button } from '@gotitinc/design-system';

const { Label } = Button;

export default ({
  label,
  variant = 'primary',
  width = 'auto',
  onClick = () => {},
  ...rest
}) => {
  return (
    <Button variant={variant} width={width} onClick={onClick} {...rest}>
      <Label>{label}</Label>
    </Button>
  );
};
