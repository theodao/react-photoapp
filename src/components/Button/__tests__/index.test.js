import React from 'react';
import { cleanup, render } from '@testing-library/react';
import { shallow } from 'enzyme';
import Button from '../';

describe('Testing button component', () => {
  afterEach(cleanup);

  it('Should match snapshot', () => {
    const { asFragment } = render(<Button />);
    expect(asFragment()).toMatchSnapshot();
  });
});
