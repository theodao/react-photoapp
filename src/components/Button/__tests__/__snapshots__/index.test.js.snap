// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing button component Should match snapshot 1`] = `
<DocumentFragment>
  <button
    class="Button u-flexInline u-justifyContentCenter u-alignItemsCenter u-textDecorationNone u-roundedMedium u-textWhite hover:u-textWhite u-backgroundPrimary hover:u-backgroundPrimaryDark u-border u-borderPrimary hover:u-textDecorationNone u-cursorPointer u-textUppercase"
  >
    <span
      class="Button-label u-inlineBlock"
    />
  </button>
</DocumentFragment>
`;
